### DISCUSSION TITLE
Examples repo with submodules

### DISCUSSION THREAD

**Discussion by iparask** [radical-cybertools/radical.pilot#2319] [Ideas]
Would it make sense to have an example repos like this: https://github.com/rapidsai/notebooks?

**Comment by andre-merzky** [comment]
So far, RP (and RCT in general) did not go well together with the cell based execution in notebooks.  Amongst others, the notebooks are not able to capture the state of the component processes we spawn in the background, and executing cells out of order screws up several assumptions our code makes, among them state transition order etc.  I mean, just *presenting* notebooks to show how things work would be possible - actually running them reliably by an end user might be tricky...

**Reply by iparask** [reply]
I was thinking more in the terms of github submodules. Sorry I forgot to mention it.

**Reply by andre-merzky** [reply]
Like listed in `https://github.com/rapidsai/notebooks/blob/branch-0.18/.gitmodules` ?  What would be the purpose of a similar RCT repo with, presumably, `radical.*` repos as git submodules?

**Reply by iparask** [reply]
Yes! We can have all the examples of RCT in a single place for users to download, as they can be centralized.

**Reply by andre-merzky** [reply]
Hmm, is that really a problem we have you are trying to address here?
